test/bats/provider.bats

```diff
@@ -0,0 +1,164 @@
+    kubectl --namespace=csi exec vault-0 -- vault kv put secret/foo2 bar2=hello2
+    kubectl --namespace=csi exec vault-0 -- vault kv put secret/foo-sync1 bar1=hello-sync1
+    kubectl --namespace=csi exec vault-0 -- vault kv put secret/foo-sync2 bar2=hello-sync2
+
+    # Create shared k8s resources.
+    kubectl create namespace test
+    kubectl --namespace=test apply -f $CONFIGS/vault-foo-secretproviderclass.yaml
+    kubectl --namespace=test apply -f $CONFIGS/vault-foo-sync-secretproviderclass.yaml
+    kubectl --namespace=test apply -f $CONFIGS/vault-foo-sync-multiple-secretproviderclass.yaml
+    } > $SETUP_TEARDOWN_OUTFILE
+}
+
+teardown(){
+    if [[ -n $SKIP_TEARDOWN ]]; then
+      echo "Skipping teardown"
+      return
+    fi
+
+    { # Braces used to redirect all teardown logs.
+    # Teardown Vault configuration.
+    kubectl --namespace=csi exec vault-0 -- vault auth disable kubernetes
+    kubectl --namespace=csi exec vault-0 -- vault policy delete example-readonly
+    kubectl --namespace=csi exec vault-0 -- vault kv delete secret/foo1
+    kubectl --namespace=csi exec vault-0 -- vault kv delete secret/foo2
+    kubectl --namespace=csi exec vault-0 -- vault kv delete secret/foo-sync1
+    kubectl --namespace=csi exec vault-0 -- vault kv delete secret/foo-sync2
+
+    # Teardown shared k8s resources.
+    kubectl delete --ignore-not-found namespace test
+    kubectl delete --ignore-not-found namespace negative-test-ns
+    } > $SETUP_TEARDOWN_OUTFILE
+}
+
+@test "1 Inline secrets-store-csi volume" {
+    kubectl --namespace=test apply -f $CONFIGS/nginx-inline-volume.yaml
+    kubectl --namespace=test wait --for=condition=Ready --timeout=60s pod/nginx-secrets-store-inline
+
+    result=$(kubectl --namespace=test exec nginx-secrets-store-inline -- cat /mnt/secrets-store/bar1)
+    [[ "$result" == "hello1" ]]
+
+    result=$(kubectl --namespace=test exec nginx-secrets-store-inline -- cat /mnt/secrets-store/bar2)
+    [[ "$result" == "hello2" ]]
+}
+
+@test "2 Sync with kubernetes secrets" {
+  # Deploy some pods that should cause k8s secrets to be created.
+  kubectl --namespace=test apply -f $CONFIGS/nginx-env-var.yaml
+  kubectl --namespace=test wait --for=condition=Ready --timeout=60s pod -l app=nginx
+
+  POD=$(kubectl --namespace=test get pod -l app=nginx -o jsonpath="{.items[0].metadata.name}")
+  result=$(kubectl --namespace=test exec $POD -- cat /mnt/secrets-store/bar1)
+  [[ "$result" == "hello-sync1" ]]
+
+  result=$(kubectl --namespace=test exec $POD -- cat /mnt/secrets-store/bar2)
+  [[ "$result" == "hello-sync2" ]]
+
+  result=$(kubectl --namespace=test get secret foosecret -o jsonpath="{.data.pwd}" | base64 -d)
+  [[ "$result" == "hello-sync1" ]]
+
+  result=$(kubectl --namespace=test exec $POD -- printenv | grep SECRET_USERNAME | awk -F"=" '{ print $2 }' | tr -d '\r\n')
+  [[ "$result" == "hello-sync2" ]]
+
+  run kubectl get secret --namespace=test foosecret
+  [ "$status" -eq 0 ]
```

Comment:
Should these be moved above lines 85-86? Since 85-86 are using the result of `get secret foosecret`, it seems like the status code should be checked before that.